### ISSUE TITLE
Iconize tabs

### ISSUE THREAD

**Issue by ReinerBforartists** [Bforartists/Bforartists#2539]
When we add icons instead of text, then we have more space for the addons. Doesn't solve the addons dilemma when you have too much addons installed. But frees nevertheless some space. And makes the look cleaner. Less visual noise. And less mouse movement.

Tabs with button works in the properties editor just fine. So i don't see a reason why it should not work in tool shelf and sidebar. Given that we manage to implement it.

Maybe we can make this one modal. Maybe we can offer text and icon too.

![icontabs](https://user-images.githubusercontent.com/13919702/124718526-77f0ce00-df06-11eb-8744-4e3e24f7b1c6.jpg)

**Comment by Draise14** [comment]
This has been requested in Rclickselect for so long.

**Comment by ReinerBforartists** [comment]
Yeah, would be nice to have. I am just not sure if we can manage it. The panels are python. The icon tabs in the Preferences are C code. But it is at the list now :)

**Comment by ReinerBforartists** [comment]
I have still no idea how to tackle this one. The tabs comes from python scripts. By the bl_category = "mytabname" tag.

**Comment by ReinerBforartists** [comment]
Hm, can be done. Somebody had a crack at it already. But the code is not public. 

https://devtalk.blender.org/t/ui-improvement-sidebar-icons-instead-of-text-labels-poc/12079

**Comment by Draise14** [comment]
I think I will give Iyad a challenge to this one later in time, but with some sponsorship. So will postpone for now.

**Comment by iyadahmed** [comment]
Kudos :pray:

**Comment by Blenux** [comment]
Finally being looked at in 2026

**Comment by ReinerBforartists** [comment]
Time files when you have fun :)

**Comment by Blenux** [comment]
This is now supported from Blender, so we can close this, create another task for which icons to be used?

<img width="149" height="344" alt="Image" src="https://github.com/user-attachments/assets/978ab7ce-db76-4e6c-b6c9-f7f61f735df9" />

**Comment by Draise14** [comment]
Do.. you have a quick example as to how to do this? Then we can open a task for the icons for the types of built in tabs. 

Also open the debate, always icons? Or only icons in compact mode?

**Comment by Blenux** [comment]
Easy to do,

bl_category = "Object"
**bl_icon = "OBJECT_DATA"**

setting the bl_icon option will display the icon in compact tabs, if none is set, it will show the 2 letter character text, alternatively one can also use

bl_icon_value = 2

but one needs to know the icon value etc.
